### PR TITLE
fix: granularity should not be zero (first PR to the team :D )

### PR DESCRIPTION
**Pull request by janne123456789**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

## Contributor Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added an example of my new feature and included it in the documentation.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** [document](https://cognitedata.github.io/indsl/contribute.html).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My Pull Request name follows the [naming convention](https://www.conventionalcommits.org/en/v1.0.0/) `fix: <description>`, `feat: <description>`, etc.

## Reviewer Checklist for Charts compliant functions:

- [ ] The docstrings of the new function follow the contributing [guidelines](https://cognitedata.github.io/indsl/contribute.html).
- [ ] The new function is professionally [documented](https://journal.emwa.org/writing-matters/whats-your-problem-a-practical-approach-to-scientific-document-design/article/2110/2047480612z2e00000000042.pdf)
- [ ] The new function and associated scripts are covered by one or more unit tests and code coverage did not decrease.
- [ ] The new function is accompanied by an example and it is included in the Gallery of Charts.
- [ ] The new function is reviewed in Chromatic. Access the storybook build results url and comment, approve or deny.
- [ ] All function inputs, arguments, and outputs have a supported data type and have human readable names.
- [ ] No code language is included in the description of the function or parameters (e.g use "polynomial order" instead of "poly_order")

### CODE REVIEW

File: indsl/validations.py [hunk 1]
```diff
@@ -35,5 +35,5 @@ def validate_timedelta(timedelta: pd.Timedelta) -> None:
     if timedelta is pd.NaT:
         raise UserValueError("Timedelta is invalid (NaT).")
 
-    if timedelta.value < 0:
-        raise UserValueError("Timedelta must be strictly positive. The smallest possible value is '1ns'")
+    if timedelta.value <= 0:
+        raise UserValueError("Timedelta must be strictly positive. The smallest possible value is '1s'")
```
Comment (tuanng-cognite) [inline indsl/validations.py:39]: I think the smallest value is still 1ns as inferred from the doc
https://pandas.pydata.org/docs/reference/api/pandas.Timedelta.html

Comment (janne123456789) [reply indsl/validations.py:39]: @tuanng-cognite I agree that the timedelta can be defined as 1ns from pandas' side, but this is the error message we give to the user when using the function resample_to_granularity, which can only handle units down to 1s (if I got it correctly). That is why I changed it.

Comment (tuanng-cognite) [reply indsl/validations.py:39]: `Valid time units are day or d, hour h, minute or m and second or s` this is for `granularity` input of `resample_to_granularity`. But `validate_timedelta` is not specific for `resample_to_granularity`, it is used in other places as well.

If we want to enforce minimum 1s, we should do that inside `resample_to_granularity`. But I don't think it is this task purpose.

Comment (janne123456789) [reply indsl/validations.py:39]: Aha, ofc - did not realize the validate_timedelta function is being used by other functions that might allow 1ns. Will update 👍

**Review by tuanng-cognite — APPROVED** [review]
good job :+1:

**Pull request merged**